server/models/contests/addGroup.js

```diff
@@ -13,30 +13,17 @@ function addGroup({ params, body, username }) {
     const { contest_id: contestId } = params
     const { group_id: groupId } = body
     pool.query(
-      `SELECT * FROM contests_moderators WHERE contest_id=? AND moderator=?`,
-      [contestId, username],
-      (error, results) => {
-        if (error) {
+      `INSERT INTO contests_groups (contest_id, group_id) SELECT contest_id, ? FROM contests_moderators WHERE contest_id=? AND moderator=?`,
+      [groupId, contestId, username],
+      (error, res) => {
+        if (error || res === undefined) {
+          const { code } = error
+          if (code === 'ER_DUP_ENTRY') {
+            return reject('The group is already added')
+          }
```

Comment:
What if the person who sent the request is not a moderator ?
In that case, the error should be returned appropriately as "You don't have moderator access to the contest"